torch/csrc/jit/script/compiler.cpp

```diff
@@ -1734,6 +1734,53 @@ struct to_ir {
     return emitSlice(loc, sliceable, maybe_dim, slice_exp);
   }
 
+  Value* emitTupleIndex(const SourceRange& loc,
+      Value * tuple_val,
+      Value * idx_val) {
+    auto throw_integer_constants_error = [loc]() {
+      throw ErrorReport(loc)
+        << "Indexing on tuples only supported with integer constants";
+    };
+
+    if (!idx_val->type()->isSubtypeOf(IntType::get())) {
+      throw_integer_constants_error();
+    }
+
+    int64_t index;
+    if (idx_val->node()->kind() == prim::Constant) {
```

Comment:
This should be using the standard ways of turning a value into a constant (see `constants.h`).
If you need neg to work, you should do constant folding in the neg operator itself, and not here.

Reply:
Why does getting the constant out of the input to aten::neg not work ?

Reply:
Because it is fragile to do it only here -- it only gets tested along this codepath, so it is more likely to be rare and wrong. It also does not benefit other locations where someone expects a -1 to be an actual constant while type checking. 

Also, this code fails to check that the nested constant is an integer, and the call to `->i` will hard-fail when it is not.